src/taipy/core/notification/notifier.py

```diff
@@ -18,20 +18,17 @@
 
 
 class Notifier:
-    # TODO add a lock on _registration
     _registrations: List[Registration] = []
```

Comment:
a dictionary would be better for adding/removing by id

Reply:
**Adding more context:** We need more design here because a single client (GUI) with a single `register_id` may register multiple topics. So maybe the name `register_id` is bad. It could be `client_id`, for example.  That also means: 
- the `register` method must behave like adding a new Registration linked to the `register_id`.
- the `unregister` method must remove all the registrations associated with the `register_id`.

**About the `_registrations` variable type:**
I initially thought about a dictionary as well. It should be a `dict[str, List[Registration]]`.
Pros: 
- That makes un-registration easier. 
Cons: 
- That makes the publication much harder. Indeed, the parsing of the registrations become much tricky, and so is the matching of the topic with the event published.

Reply:
We will go with a dictionary (key is topics, value is a list of registration object) due to huge benefit in matching topics and events while un-registration for a subscriber is expected to be not so complicated